app/src/main/java/com/travelguide/activities/TravelGuideActivity.java

```diff
@@ -46,6 +143,25 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
     @Override
     public boolean onCreateOptionsMenu(Menu menu) {
         getMenuInflater().inflate(R.menu.menu_travel_guide_activity, menu);
+        MenuItem searchItem = menu.findItem(R.id.action_search);
+        final SearchView searchView = (SearchView) MenuItemCompat.getActionView(searchItem);
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                if (query != null)
```

Comment:
You can use this method: http://developer.android.com/reference/android/text/TextUtils.html#isEmpty(java.lang.CharSequence)